From 74406e5ea3378a9b633ad0776a4bb34347cf1911 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sat, 7 Dec 2013 19:45:45 -0800
Subject: [PATCH] Copy root field from wire event into root, not window, of
 Xlib event

Fixes cppcheck warning:
[src/XScrnSaver.c:104] -> [src/XScrnSaver.c:105]: (performance, inconclusive)
 Variable 'window' is reassigned a value before the old one has been
 used if variable is no semaphore variable.

Error seems to date back to revision 1.1 in X Consortium RCS,
dated 1992-02-13, by "keith"

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
Reviewed-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/XScrnSaver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/XScrnSaver.c b/src/XScrnSaver.c
index 2a89246..3947dd5 100644
--- a/src/XScrnSaver.c
+++ b/src/XScrnSaver.c
@@ -102,7 +102,7 @@ static Bool wire_to_event (
     	se->send_event = (sevent->type & 0x80) != 0;
     	se->display = dpy;
     	se->window = sevent->window;
-    	se->window = sevent->root;
+	se->root = sevent->root;
     	se->state = sevent->state;
 	se->kind = sevent->kind;
 	se->forced = True;
-- 
2.8.3

